### PR TITLE
Makes stuttering actually legible

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Makes stuttering actually legible, it now has the chance to stutter the first letter of a word, and which makes stuttering actually readable and not just someone griefing when they get it roundstart as a gene. The base code that the new stutter comes from is SS13-Colonial Marines.

The chance to stutter is 25% to 50% on each word, but has a minimum bound of your stamina loss. Are you in stam-crit? You're gonna stutter 100% of the time.

## Why It's Good For The Game

Being able to roleplay with others in a roleplay game is good.

## Images of changes

image

Heres it on an organic, and on a machine.
image

## Testing

See above

## Changelog
:cl:
tweak: Stuttering has been changed to be more readable, now only affecting the beginning of words.
/:cl:

### CODE REVIEW

File: code/modules/mob/mob_misc_procs.dm [hunk 1]
```diff
@@ -273,50 +273,38 @@
...
-					else
-						n_letter = text("[n_letter]-[n_letter]")
-		t = text("[t][n_letter]")//since the above is ran through for each letter, the text just adds up back to the original word.
-		p++//for each letter p is increased to find where the next letter will be.
-	return sanitize(copytext(t,1,MAX_MESSAGE_LEN))
+/proc/stutter(phrase, stamina_loss = 0, robotic = FALSE)
+	phrase = html_decode(phrase)
+
+	var/list/split_phrase = splittext(phrase, " ") //Split it up into words.
+
```
Comment (1080pCat) [inline code/modules/mob/mob_misc_procs.dm:280]:
```suggestion

```

File: code/modules/mob/mob_misc_procs.dm [hunk 2]
```diff
@@ -273,50 +273,38 @@
...
+			first_letter = pick("@", "!", "#", "$", "%", "&", "?")
+			if(prob(25))
+				second_repeat = pick("@", "!", "#", "$", "%", "&", "?")
+		//Repeat the first letter to create a stutter.
+		if(rand(1, 3) == 1) // more accurate than prob(33.333333)
+			word = "[first_letter]-[second_repeat]-[word]"
+		else
+			word = "[first_letter]-[word]"
+
+		split_phrase[index] = word
```
Comment (1080pCat) [inline code/modules/mob/mob_misc_procs.dm:306]:
```suggestion

```
this one im not awfully sure about, just kinda seems like unnecessary white space to me

File: code/modules/mob/mob_misc_procs.dm [hunk 3]
```diff
@@ -273,50 +273,38 @@
...
+		if(robotic && prob(50))
+			first_letter = pick("@", "!", "#", "$", "%", "&", "?")
+			if(prob(25))
+				second_repeat = pick("@", "!", "#", "$", "%", "&", "?")
+		//Repeat the first letter to create a stutter.
+		if(rand(1, 3) == 1) // more accurate than prob(33.333333)
+			word = "[first_letter]-[second_repeat]-[word]"
+		else
+			word = "[first_letter]-[word]"
+
```
Comment (1080pCat) [inline code/modules/mob/mob_misc_procs.dm:304]:
```suggestion

```
same here tbh

File: code/modules/mob/mob_misc_procs.dm (hunk 1 above)
Comment (Contrabang) [reply code/modules/mob/mob_misc_procs.dm:279]: I dont fully agree with all of these, but ill minimize the whitespace

**Review by Henri215 — APPROVED** [review]
LGTM

**Review by 1080pCat — APPROVED** [review]

**Review by S34NW — APPROVED** [review]
